### ISSUE TITLE
RAM consumption is 3 - 4 times the cached data

### ISSUE THREAD

**Issue by schopdekar** [cornerstonejs/cornerstone3D#805]
### Describe the Bug

Hi team,

When I am trying to render a data of size 300mb using volume viewport the RAM consumption in the activity monitor seems to be 1.2gb which is 3 -4 times.

Troubleshooting on volumeBasic Example:
1. Commented out the code from createAndCacheVolume function call. Found that the RAM consumption was minimal as volume is not yet loaded in cache.
![image](https://github.com/cornerstonejs/cornerstone3D/assets/134197565/3afc9e38-9728-481c-a2f6-58e556337d41)

2. Uncomment createAndCacheVolume function and kept below code commented which is having load and setVolumes function. Found that the RAM consumption was as per the volume size as the volume is now loaded in the cache.
![image](https://github.com/cornerstonejs/cornerstone3D/assets/134197565/34067c36-1f59-4d56-a526-21a0f5297ea8)

3. Now uncommented load function and found that the RAM consumption incremented by data size
![image](https://github.com/cornerstonejs/cornerstone3D/assets/134197565/a3d98b71-453d-42f3-a3c3-d7e04cd213d6)

4. Now uncommented setVolumes function and found that the RAM consumption incremented by data size
![image](https://github.com/cornerstonejs/cornerstone3D/assets/134197565/db8ffa37-b5ad-401b-894e-3db2e805c26f)

### Steps to Reproduce

1. Use a data-set of size 300mb.
2. Use the volumeBasic example and run on safari/chrome.
3. Monitor the RAM usage in activity monitor.

### The current behavior

RAM consumption seems to be very high than the actual data when monitored.

### The expected behavior

RAM consumption should be as per the data size.

### OS

Windows, Mac OS

### Node version

v16.15.0

### Browser

Chrome: 117.0.5938.92, Safari: 17.0

**Comment by Hamusuta0320** [comment]
Yes, I have also encountered this issue. The JavaScript heap occupies 500m, but the Chrome task manager shows that it occupies over two gigabytes of memory. @sedghi

**Comment by Hamusuta0320** [comment]
> size

Thank you very much for your answer, solved my memory problems some questions, I have another question, MPR how to support more than one thousand, more than two thousand images in series, when I try to view the MPR of this order of magnitude sequence, it will prompt that the array cannot be allocated, this need to view a large number of pictures  MPR scene how to deal with?

**Comment by sedghi** [comment]
It is not a matter of number of slices, it is the size of eventual monitor
If you can't fit it regularly you can have two options in the configuration of `init`

`preferSizeOverAccuracy`: as the name suggest it will make the size smaller using halfFloat (11 bit, not 16 bit) but as a matter of reducing accuracy 

Integers between 0 and 2048 can be exactly represented (and also between −2048 and 0)
Integers between 2048 and 4096 round to a multiple of 2 (even number)
Integers between 4096 and 8192 round to a multiple of 4
Integers between 8192 and 16384 round to a multiple of 8
Integers between 16384 and 32768 round to a multiple of 16
Integers between 32768 and 65519 round to a multiple of 32

`useNorm16Texture`: reduces the size (in half) with full accuracy, but has some issues in some devices

**Comment by Hamusuta0320** [comment]
> It is not a matter of number of slices, it is the size of eventual monitor
> If you can't fit it regularly you can have two options in the configuration of `init`
> 
> `preferSizeOverAccuracy`: as the name suggest it will make the size smaller using halfFloat (11 bit, not 16 bit) but as a matter of reducing accuracy 
> 
> Integers between 0 and 2048 can be exactly represented (and also between −2048 and 0)
> Integers between 2048 and 4096 round to a multiple of 2 (even number)
> Integers between 4096 and 8192 round to a multiple of 4
> Integers between 8192 and 16384 round to a multiple of 8
> Integers between 16384 and 32768 round to a multiple of 16
> Integers between 32768 and 65519 round to a multiple of 32
> 
> `useNorm16Texture`: reduces the size (in half) with full accuracy, but has some issues in some devices
> 

Thank you very much for the two solutions you provided, which gave me a better understanding of this problem. I will try to use these two methods later and see the results. That is to say, the high usage of memory space is due to a bug in using 16 bit groups to store data. Therefore, using 32-bit arrays to store 16 bit arrays results in more memory usage, so this is not a problem with the number of sequences. If my understanding is correct

**Comment by schopdekar** [comment]
Hey @sedghi 
When tried with unit8 data we found the same RAM problem.

On further troubleshooting with you in the OHIF weekly meeting we found that the web workers are causing the extra RAM consumption.

Can you please let us know what can be a solid solution to resolve this issue?
